evalml/pipelines/utils.py

```diff
@@ -163,6 +163,19 @@ class TemplatedPipeline(_get_pipeline_base_class(problem_type)):
     return TemplatedPipeline({c.name: c.parameters for c in component_instances}, random_state=random_state)
 
 
+def _json_dict(dictionary):
+    """Helper method to make sure all elements in the provided dictionary are JSON-serializable
+    """
+    for v in dictionary.values():
+        if isinstance(v, dict):
+            _json_dict(v)
+        else:
+            try:
+                json.dumps(v)
+            except TypeError:
+                raise TypeError(f"Value {v} of type {type(v)} cannot be JSON-serialized")
```

Comment:
@bchen1116 for what types does this error occur? Should we add test coverage of those cases?

Reply:
It errors out for lots of python-specific types. I can add more coverage of that in the tests